docs/python/1.6/python-appapi-devguide-5.md

```diff
@@ -67,3 +77,181 @@ def main():
+* a explicit PE restart request,
+* a parallel region width change (IBM Streams 4.3 or later)
+
+You can use a consistent region in your stream processing applications to avoid data loss due to software or hardware failure and meet your requirements for at-least-once processing.
+
+The application or a portion of it may be configured to maintain state after a PE restart by the mechanism **consistent region**.
+A **consistent region** is a subgraph where the states of callables become consistent by processing all the tuples at least once within defined points on a stream.
+On a failure, a region is reset to its last successfully persisted state, and source callable of a region can replay any tuples submitted after the restored state.
+
+![cr_reset](/streamsx.documentation/images/python/cr_reset.png)
+
+
+The consistent region feature requires a configured [checkpoint repository](https://www.ibm.com/support/knowledgecenter/SSCRJU_5.3/com.ibm.streams.cfg.doc/doc/ibminfospherestreams-configuring-checkpoint-data-store.html) in the Streams instance and Streams applications must be submitted in [DISTRIBUTED](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.context.html#streamsx.topology.context.ContextTypes.DISTRIBUTED) context.
+
+This section will discuss how to create stream processing applications with the consistent region feature:
+
+* [Configure a consistent region](#configure)
+* [Stateful callables](#stateful)
+* [Use SPL operators supporting consistent region](#spl_cr)
+
+
+<a id="configure"></a>
+## Configure a consistent region
+
+* [streamsx.topology.topology.Stream.set_consistent()](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.topology.html#streamsx.topology.topology.Stream.set_consistent)
+* [ConsistentRegionConfig](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.state.html#streamsx.topology.state.ConsistentRegionConfig)
+
+The recommended way to create a `ConsistentRegionConfig` is to call either `operator_driven()` or `periodic()`.
+
+For example, set a source stream `s` to be a the start of an operator driven consistent region with a drain timeout of five seconds and a reset timeout of twenty seconds:
+```
+s.set_consistent(ConsistentRegionConfig.operatorDriven(drain_timeout=5, reset_timeout=20))
+```
+
+Example of a periodic consistent region configuration, IBM Streams runtime will trigger a drain and checkpoint every 30 seconds:
+```
+s.set_consistent(ConsistentRegionConfig.periodic(period=30, drain_timeout=40, reset_timeout=40, max_consecutive_attempts=6))
+```
+
+Reference
+* [streamsx.topology.state](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.state.html#module-streamsx.topology.state)
+
+<a id="stateful"></a>
+## Stateful callables
+
+Use of a class instance allows a transformation to be stateful by maintaining state in instance attributes across invocations.
+
+When the callable is in a consistent region then it is serialized using dill.
```

Comment:
omit the word "then" in the sentence _When the callable is in a consistent region then it is serialized using dill._